hooks/gen_docs.py

```diff
@@ -1,13 +1,359 @@
+            elif not env_var.default_value:
+                default_value = f" {COMMENT_SYMBOL} No default value, not required"
+            else:
+                default_value = f"={env_var.default_value}"
+            env_var.default_value = default_value
+            with target.open("a") as f_target:
+                f_target.write(
+                    f"{env_var.description}\n"
+                    f"{env_var.name}{env_var.default_value}\n",
+                )
+
+
+def write_csv_to_md_file(
+    source: Path,
+    target: Path,
+    title: str | None,
+    description: str | None = None,
+    heading: str = "###",
+) -> None:
+    """Write csv data from a file into a markdown file.
+
+    :param source: path to csv file to read from
+    :param target: path to md file to overwrite or create
+    :param title: Title for the table, optional
+
+    """
+    with target.open("w+") as f:
+        if title:
+            f.write(f"{heading} {title}\n")
+        if description:
+            f.write(f"\n{description}\n\n")
+        writer = MarkdownTableWriter()
+        writer.from_csv(str(source))
+        writer.table_name = ""
+        writer.dump(output=f)
+
+
+# copy examples from tests resources
+shutil.copyfile(
+    PATH_ROOT / "tests/pipeline/resources/component-type-substitution/pipeline.yaml",
+    PATH_DOCS_VARIABLES / "variable_substitution.yaml",
+)
+
+# Descriptions for the tables and the dotenv files are the same, except for the
+# sentence bellow
+DESCRIPTION_ADDITION_DOTENV_FILE = "The default setup is shown. "
+
+# find all config-related env variables and write them into a file
+PATH_CONFIG_ENV_VARS_DOTENV_FILE = PATH_DOCS_VARIABLES / "config_env_vars.env"
+PATH_CONFIG_ENV_VARS_MD_FILE = PATH_DOCS_VARIABLES / "config_env_vars.md"
+PATH_CONFIG_ENV_VARS_CSV_FILE = PATH_DOCS_VARIABLES / "temp_config_env_vars.csv"
+CONFIG_ENV_VARS_TITLE = "Pipeline config environment variables"
+CONFIG_ENV_VARS_DESCRIPTION = (
+    "These variables are a lower priority alternative to the settings in `config.yaml`. "
+    "Variables marked as required can instead be set in the pipeline config."
+)
+# Overwrite the temp csv file
+with PATH_CONFIG_ENV_VARS_CSV_FILE.open("w+") as f:
+    csv.writer(f).writerow(
+        [
+            EnvVarAttrs.NAME,
+            EnvVarAttrs.DEFAULT_VALUE,
+            EnvVarAttrs.REQUIRED,
+            EnvVarAttrs.DESCRIPTION,
+            EnvVarAttrs.CORRESPONDING_SETTING_NAME,
+        ],
+    )
+write_title_to_dotenv_file(
+    PATH_CONFIG_ENV_VARS_DOTENV_FILE,
+    CONFIG_ENV_VARS_TITLE,
+    DESCRIPTION_ADDITION_DOTENV_FILE + CONFIG_ENV_VARS_DESCRIPTION,
+)
+# NOTE: This does not see nested fields, hence if there are env vars in a class like
+# TopicConfig(), they wil not be listed. Possible fix with recursion.
+config_fields = PipelineConfig.__fields__
+for config_field_name, config_field in config_fields.items():
+    config_field_info = PipelineConfig.Config.get_field_info(config_field.name)
```

Comment:
what's the difference between `config_field_name` and `config_field.name`?

Reply:
There is no difference, nice catch!

Reply:
Fixed it